extensions/infinispan-client/deployment/src/main/java/io/quarkus/infinispan/client/deployment/devservices/InfinispanDevServiceProcessor.java

```diff
@@ -259,7 +259,7 @@ private static class QuarkusInfinispanContainer extends InfinispanContainer {
 
         public QuarkusInfinispanContainer(String clientName, InfinispanDevServicesConfig config,
                 LaunchMode launchMode, boolean useSharedNetwork) {
-            super(config.imageName.orElse(IMAGE_BASENAME + ":" + Version.getMajorMinor()));
+            super(config.imageName.orElse(IMAGE_BASENAME + ":" + Version.getVersion()));
```

Comment:
@karesti you also changed this in this patch. What was the rationale? It doesn't look strictly related and I want to make sure that's something we want in 3.8.

Reply:
getMajorMinor will retrieve 14.0 while getVersion will retrieve 14.0.16.Final image
We update images to map the latest minor. However, if the user has the image locally (14.0) won't get updated. There might be bugs that are fixed on the server that will still be present in the local machine of the user for this reason. Forcing with getVersion the dev service container image, we guarantee the user is running the same version locally and on the server.

Reply:
OK, I think the rationale makes sense and we can backport it too.

@rsvoboda WDYT?

Reply:
Agree, let's have it @gsmet.

Reply:
So this is a problem in the end, because on product side we end up with something like `quay.io/infinispan/server:14.0.27.Final-redhat-00001`

I guess we need to get back to `super(config.imageName.orElse(IMAGE_BASENAME + ":" + Version.getMajorMinor()));` even for main because that would be a problem for the next feature release

Reply:
@rsvoboda I need to check with the team, but for upstream this is the best way for dev services, otherwise during N minor versions, fixes and updates, unless we always pull the image locally, it will fail. For downstream I need to check

Reply:
@rsvoboda I understand the issue, I'll change this back but I need to figure out how to avoid this problem in Quarkus upstream